modules/preview-react/pill/lib/usePillModel.tsx

```diff
@@ -0,0 +1,46 @@
+import {createEventMap, Model, ToModelConfig, useEventMap} from '@workday/canvas-kit-react/common';
+
+type PillState = {
+  maxWidth?: number | string;
+  /**
+   * Use to determine if a pill is of type `button` and pass the click event to the semantic button element
+   */
+  onClick?: () => void | undefined;
+  onDelete?: () => void | undefined;
+  disabled?: boolean;
+};
+
+type PillEvents = {};
+
+export type PillModel = Model<PillState, PillEvents>;
+
+const pillEventMap = createEventMap<PillEvents>()({
+  guards: {},
+  callbacks: {},
```

Comment:
I think moving `onClick` and `onDelete` to events will also affect this config as well.